### ISSUE TITLE
X11 session fails to start on Ubuntu Jammy

### ISSUE THREAD

**Issue by kgilmer** [regolith-linux/regolith-session#36]
This issue is found in `unstable`, containing the latest package versions.

STR:
1. Install regolith x11 session into Ubuntu Jammy 22.04.04
2. Log into Regolith X11 session

Expected: see desktop
Actual: "error occurred", returns to login screen

Attached log file does not seem to provide a smoking gun, but that for some reason the x11 session is being started multiple times.  Wondering if @SoumyaRanjanPatnaik has any thoughts on how to further troubleshoot.

**Comment by kgilmer** [comment]
[log.txt](https://github.com/user-attachments/files/15859309/log.txt)

Log snippet at time of session init

**Comment by kgilmer** [comment]
Posit of multiple instances of i3 being launched from this log line:

```
Jun 16 10:54:38 testy-ubuntu-jammy regolith-session-x11[2077]: 06/16/2024 10:54:38 AM - [libi3] ERROR: Refusing to create UNIX socket at /run/user/1000/i3/ipc-socket.1949: Socket is already in use
```

**Comment by kgilmer** [comment]
Issue seems only to be present on Jammy.  Noble X11 and Sway sessions via `lightdm` and `gdm3` start successfully.

**Comment by SoumyaRanjanPatnaik** [comment]
> Posit of multiple instances of i3 being launched from this log line:
> 
> ```
> Jun 16 10:54:38 testy-ubuntu-jammy regolith-session-x11[2077]: 06/16/2024 10:54:38 AM - [libi3] ERROR: Refusing to create UNIX socket at /run/user/1000/i3/ipc-socket.1949: Sock eet is already in use
> ```

I see this line in the logs indicating i3 being started by `regolith-x11.desktop`, with pid 1894.  
```
Jun 16 10:54:38 testy-ubuntu-jammy regolith-x11.desktop[1894]: Regolith is launching i3 with /etc/regolith/i3/config
```

Later, `regolith-session-x11` seems to start i3, this time with pid 2050. 
```
Jun 16 10:54:38 testy-ubuntu-jammy regolith-session-x11[2050]: Regolith is launching i3 with /etc/regolith/i3/config
```

Its pretty clear there are two seperate instances of i3 being started.

**Comment by SoumyaRanjanPatnaik** [comment]
I think I've found the problem:

On jammy, session management is not delegated to `systemd` and should be handled by `gnome-session` due to the `--builtin` . `systemd` should only be used on `bionic`, `buster`, `noble` and `trixie`. However, the logs seem to indicate that `systemd` is also for some reason starting the session.

```
Jun 16 10:54:38 testy-ubuntu-jammy systemd[1417]: Starting Regolith Session for i3...
```

This means, both `gnome-session` and `systemd` are trying to start the session, casuing a race condition between both the sessions. The one that starts i3 first will succeed in acquiring the socket and the second instance will fail.

**Comment by SoumyaRanjanPatnaik** [comment]
Possible fix in #37. 

The target file for both the sessions had a `Wants` directive that tried to start the session via systemd. This means, [starting target files](https://github.com/regolith-linux/regolith-wm-config/pull/28/files) causes the systemd to start the corresponding services, leading to double session start.
